stream-chat-android-offline/src/main/java/io/getstream/chat/android/offline/querychannels/NonMemberChatEventHandler.kt

```diff
@@ -0,0 +1,108 @@
+package io.getstream.chat.android.offline.querychannels
+
+import io.getstream.chat.android.client.ChatClient
+import io.getstream.chat.android.client.api.models.FilterObject
+import io.getstream.chat.android.client.events.ChannelUpdatedByUserEvent
+import io.getstream.chat.android.client.events.ChannelUpdatedEvent
+import io.getstream.chat.android.client.events.MemberAddedEvent
+import io.getstream.chat.android.client.events.MemberRemovedEvent
+import io.getstream.chat.android.client.events.NotificationAddedToChannelEvent
+import io.getstream.chat.android.client.events.NotificationMessageNewEvent
+import io.getstream.chat.android.client.events.NotificationRemovedFromChannelEvent
+import io.getstream.chat.android.client.models.Channel
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.runBlocking
+
+/**
+ * Implementation of [ChatEventHandler] that handles events when ChannelListViewModel has a filter to show channels which
```

Comment:
I would avoid here mentioning `ChannelListViewModel`. Because this module is not aware of any UI part. It's better to substitute it by `a query channels request has a filter to include channels where the current user is not a member`